script.js

```diff
@@ -1,12 +1,105 @@
 //You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+let arrayOfEpisodes = getAllEpisodes();
+let mainEl = document.querySelector("main");
+let searchBar = document.querySelector("#searchInput");
+let displayNumberOfEpisodes = document.querySelector("#displayEpisodesText");
+let selectEl = document.querySelector("#selectMenu");
+
+//search bar
+searchBar.addEventListener("keyup", (e) => {
+  const searchString = e.target.value.toLowerCase();
+  const filteredEpisodes = arrayOfEpisodes.filter((episode) => {
+    return (
+      episode.summary.toLowerCase().includes(searchString) ||
+      episodeCode(episode).toLowerCase().includes(searchString) ||
+      episode.name.toLowerCase().includes(searchString)
+    );
+  });
+  loadEpisodes(filteredEpisodes);
+  displayNumberOfEpisodes.innerText = `Displaying ${filteredEpisodes.length}/${arrayOfEpisodes.length} episodes`;
+  selectMenu(filteredEpisodes);
+});
+
+//add functionality to select menu
+selectEl.addEventListener("change", (e) => {
+  window.location = `#${e.target.value}`;
+  let allEpisodes = document.getElementsByClassName("episodeBox");
+  let highlightedEl = document.getElementById(e.target.value);
+  highlightedEl.style.border = "solid";
+  for (let i = 0; i < allEpisodes.length; i++) {
+    if (allEpisodes[i] !== highlightedEl) {
+      allEpisodes[i].style.border = null;
+    }
+  }
+});
+
+//Formatting name of an episode
+function episodeCode(obj) {
```

Comment:
Have a look at JavaScript function padStart()
So that your code is flexible enough to manage if there were more than a 100 episodes

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart